python/mrtrix3/commands/5ttgen/msmt.py

```diff
@@ -0,0 +1,112 @@
+
+
+def execute(): # pylint: disable=unused-variable
+  # Extract l=0 term from WM
+  run.command('mrconvert {app.ARGS.odf_wm} -coord 3 0 -axes 0,1,2 wm_vol.mif')
+
+  # check whether the GM and CSF ODF voxel grid match that of wm_vol.mif. If the do not match, regrid with wm_vol.mif as template
+  command_size_WModf = 'mrinfo -spacing wm_vol.mif'
+  spacing_size_WModf = subprocess.check_output(command_size_WModf, shell=True).decode('utf-8')
+  command_size_GModf = 'mrinfo -spacing {app.ARGS.odf_gm}'
+  spacing_size_GModf = subprocess.check_output(command_size_GModf, shell=True).decode('utf-8')
+  command_size_CSFodf = 'mrinfo -spacing {app.ARGS.odf_gm}'
+  spacing_size_CSFodf = subprocess.check_output(command_size_CSFodf, shell=True).decode('utf-8')
+
+  if spacing_size_WModf == spacing_size_GModf:
+    print("voxel grid of GM and WM ODF match")
+    run.command('mrconvert {app.ARGS.odf_gm} gm_vol.mif')
+  else:
+    print("WARNING: GM ODF has different dimeonsions to WM ODF. Regridding GM ODF")
+    run.command('mrgrid {app.ARGS.odf_gm} -template wm_vol.mif regrid gm_vol.mif')
+
+  if spacing_size_WModf == spacing_size_CSFodf:
+    print("voxel grid of CSF and WM ODF match")
+    run.command('mrconvert {app.ARGS.odf_csf} csf_vol.mif')
+  else:
+    print("WARNING: CSF ODF has different dimeonsions to WM ODF. Regridding CSF ODF")
+    run.command('mrgrid {app.ARGS.odf_csf} -template wm_vol.mif regrid csf_vol.mif')
+
+  # Set negative values to zero
+  run.command('mrcalc wm_vol.mif 0.0 -max wm_vol_pos.mif')
+  run.command('mrcalc gm_vol.mif 0.0 -max gm_vol_pos.mif')
+  run.command('mrcalc csf_vol.mif 0.0 -max csf_vol_pos.mif')
+
+  # Normalize each volume
+  run.command('mrmath wm_vol_pos.mif gm_vol_pos.mif csf_vol_pos.mif sum totalvol.mif')
+  run.command('mrcalc wm_vol_pos.mif totalvol.mif -divide wm_vol_pos_norm.mif')
+  run.command('mrcalc gm_vol_pos.mif totalvol.mif -divide gm_vol_pos_norm.mif')
+  run.command('mrcalc csf_vol_pos.mif totalvol.mif -divide csf_vol_pos_norm.mif')
+
+  # Create empty volume for SGM and pathology
+  run.command('mrcalc wm_vol.mif inf -gt empty_vol.mif')
+
+  # Concatenate volumes
+  run.command('mrcat -datatype float32 gm_vol_pos_norm.mif empty_vol.mif wm_vol_pos_norm.mif csf_vol_pos_norm.mif empty_vol.mif fTT_dirty.mif')
+
+  #########################
+  # Tidy image by masking #
+  #########################
+
+# if mask is provided, use it. If not, create one
+  if app.ARGS.mask:
+    # case 1: using provided brainmask
+    # check if dimensions of brainmask match 5TT_dirty. If not, regrid
+    command_fTTdirty = 'mrinfo -spacing fTT_dirty.mif'
+    spacing_fTTdirty = subprocess.check_output(command_fTTdirty, shell=True).decode('utf-8')
+
+    command_mask = 'mrinfo -spacing {app.ARGS.mask}'
+    spacing_mask = subprocess.check_output(command_mask, shell=True).decode('utf-8')
+
+      if spacing_fTTdirty == spacing_mask:
+        print("mask has equal dimensions to 5TT image. No regridding of mask required")
```

Comment:
```suggestion
        app.console("mask has equal dimensions to 5TT image. No regridding of mask required")
```

Note second instance below.